### PR TITLE
[SPARK-37836][PYTHON][INFRA] Enable F841, E722, E305 and E226 for PEP 8 compliance

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to enable flake8 rules as below:

- [F841](https://www.flake8rules.com/rules/F841.html): Local variable name is assigned to but never used
- [E722](https://www.flake8rules.com/rules/E722.html): Do not use bare except, specify exception instead
- [E305](https://www.flake8rules.com/rules/E305.html): Expected 2 blank lines after end of function or class
- [E226](https://www.flake8rules.com/rules/E226.html):  Missing whitespace around arithmetic operator

We should probably still enable the rules below:

- [E731](https://www.flake8rules.com/rules/E731.html): Do not assign a lambda expression, use a def
- [E741](https://www.flake8rules.com/rules/E741.html): Do not use variables named 'I', 'O', or 'l' 
- [W503](https://www.flake8rules.com/rules/W503.html): Line break occurred before a binary operator
- [W504](https://www.flake8rules.com/rules/W504.html): Line break occurred after a binary operator

but the rules ^ are not enabled in this PR because:
- There are too many instances to fix. Maybe it's better to separate PRs
- Some require real code changes.
- W503 and W504 are sort of conflicted. We should investigate a bit more.

### Why are the changes needed?

To comply PEP 8.

### Does this PR introduce _any_ user-facing change?

No, dev-only.

### How was this patch tested?

I manually tested it Python linter.

### CODE REVIEW

File: dev/.gitignore [hunk 1]
```diff
@@ -1,2 +0,0 @@
-pep8*.py
```
Comment (HyukjinKwon) [inline dev/.gitignore:1]: We don't need it anymore after https://github.com/apache/spark/pull/34655.

File: python/pyspark/tests/test_readwrite.py [hunk 2]
```diff
@@ -231,9 +231,10 @@ def test_newhadoop(self):
         )
         self.assertEqual(result, data)
 
-        fmt = "org.apache.hadoop.mapreduce.lib.output.SequenceFileOutputFormat"
```
Comment (HyukjinKwon) [inline python/pyspark/tests/test_readwrite.py:234]: this is technically a followup of https://github.com/apache/spark/pull/35126 but just decided to piggy-back here.

**Comment by HyukjinKwon** [comment]
cc @zero323, @itholic, @ueshin, @xinrong-databricks FYI. I enabled rules that sort of need not so much changes, and that looks strait forward to me.

**Review by dongjoon-hyun — COMMENTED** [review]
Is there a merge order for this PR and the black formatter PR (https://github.com/apache/spark/pull/35127)?

It looks like both might cause some conflicts if we don't rebase the second one.

**Comment by HyukjinKwon** [comment]
Either one can be merged first, and the other one might have to be rebased. But there won't be real complicated conflicts. Just checked the diff by manually cherry-picking.

**Comment by zero323** [comment]
E305, E226 should be covered by black, if enabled for particular path, but there is no harm in adding them.

Personally, I have mixed feelings about E722 ‒ using `BaseException` or `Exception` doesn't really seem to add anything. But I guess there is no harm in that either :)

**Review by zero323 — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master. Thanks guys!

**Pull request closed without merging**